src/HttpHeader.cc

```diff
@@ -423,6 +429,8 @@ HttpHeader::parse(const char *header_start, size_t hdrLen, Http::ContentLengthIn
                     for (const char *p = this_line; p < field_end && cr_only; ++p) {
                         if (*p != '\r')
                             cr_only = false;
+                        else
+                            hasBareCr = "bare-CR";
                     }
```

Comment:
Not quite. This loop detects when the entire line consist only of a series of CR*LF. If cr_only becomes true there is a rejection immediately due to that. Otherwise it has found a pattern of CR* foo CRLF.

That prefix CR is bare-CR so does meet our policy requirement. Technically we might use this to short-circuit the memchr() test later as we already know bare-CR exists, but I have not gone that far since common case will not get here anyway.

Reply:
> Alex: Please double check this change. AFAICT, it is buggy and should be removed.
> Amos: Not quite.

The change was buggy. Thank you for removing it. I am resolving this change request.

As for your explanation of what the cr_only loop does, I believe the "Otherwise it has found a pattern of CR foo CRLF" and "prefix CR" parts are inaccurate or misleading, but perhaps you were talking about some other loop/code, so I am not going to spend time on discussing those details in this now-addressed change request.